superset/models/core.py

```diff
@@ -381,25 +383,19 @@ def get_sqla_engine_with_context(
             ssh_tunnel.bind_host = url.host
             ssh_tunnel.bind_port = url.port
             ssh_params = ssh_tunnel.parameters()
-            try:
-                with sshtunnel.open_tunnel(**ssh_params) as server:
-                    yield self._get_sqla_engine(
-                        schema=schema,
-                        nullpool=nullpool,
-                        source=source,
-                        ssh_tunnel_server=server,
-                    )
-            except Exception as ex:
-                raise ex
+            if ssh_tunnel_manager and is_feature_enabled("SSH_TUNNELING"):
+                ssh_params = ssh_tunnel_manager.mutate(ssh_params)
 
-        else:
-            # do look up in table for using database_id
-            try:
+            with sshtunnel.open_tunnel(**ssh_params) as server:
```

Comment:
This is a little confusing because it seems like you're still opening a tunnel even if the feature flag isn't enabled. I expect that it likely checks again for the feature flag downstream, but it doesn't read that way here.